src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,164 @@
+            if (semver.gte(pyVersion, minReqVersion)) {
+                this.pythonAlias = pyAlias;
+            }
+        } catch {
+            // ignore error, not installed under this alias
+        }
+    }
+
+    private async setVirtualEnviornment(): Promise<void> {
+        const funcEnv: string = 'func_env';
+        let pythonVenvPath: string;
+        const newButton: MessageItem = { title: 'Create New' };
+        const existingButton: MessageItem = { title: 'Use Existing' };
+        const venvRequired: string = localize('venvRequired', 'You must be running in a Python virtual environment to create a Python Function project. Create a new one or use an existing?');
+        const input: MessageItem = await ext.ui.showWarningMessage(venvRequired, { modal: true }, newButton, existingButton);
+        if (input === newButton) {
+            // if there is no func_env, create one as it's required for Python functions
+            await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, this.pythonAlias, '-m', 'venv', 'func_env');
+            pythonVenvPath = funcEnv;
+        } else {
+            const options: OpenDialogOptions = {
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                defaultUri: workspace.workspaceFolders && workspace.workspaceFolders.length > 0 ? workspace.workspaceFolders[0].uri : undefined,
+                openLabel: localize('select', 'Select')
+            };
+            const venvFspath: string = (await ext.ui.showOpenDialog(options))[0].fsPath;
+            pythonVenvPath = path.relative(this.functionAppPath, venvFspath);
+        }
+        await updateWorkspaceSetting(this.pythonVenvPathSetting, pythonVenvPath, this.functionAppPath);
+    }
+
+    private async getVenvActivatePath(platform: Platform): Promise<string> {
+        const venvPath: string | undefined = getFuncExtensionSetting(this.pythonVenvPathSetting, this.functionAppPath);
+        if (venvPath) {
+            switch (platform) {
+                case Platform.Windows:
+                    return path.join('.', venvPath, 'Scripts', 'activate');
+                case Platform.MacOS:
+                default:
+                    // assuming OSX and Linux use the same path -- need to some testing
+                    return path.join('.', venvPath, 'bin', 'activate');
+            }
+
+        } else {
+            // if there is no venv path, prompt user and recursively call function again
+            await this.setVirtualEnviornment();
+            return await this.getVenvActivatePath(platform);
+        }
+    }
+
+    private async createPythonProject(): Promise<void> {
+        switch (process.platform) {
+            case Platform.Windows:
+                await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, `${await this.getVenvActivatePath(Platform.Windows)} && func init ./ --worker-runtime python`);
```

Comment:
nit: can you make a constant for `func init ./ --worker-runtime python` since it's used twice